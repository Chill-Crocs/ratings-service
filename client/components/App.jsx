/* eslint-disable no-console */
import React, { Component } from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';

import getRandomItem from '../helpers/helpers.js';
import ReviewList from './ReviewList';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      reviews: [],
      start: 0,
      end: 4,
    };
    this.getReviews = this.getReviews.bind(this);
  }

  componentDidMount() {
    this.getReviews();
  }

  getReviews() {
    axios.get('/api/reviews')
      .then((response) => {
        const items = response.data;
        const reviews = getRandomItem(items);
        this.setState({
          reviews: reviews.shopReviews,
        });
      });
  }

  render() {
    const { reviews, start, end } = this.state;
    const fourReviews = reviews.slice(start, end);
    console.log(fourReviews);
    return (
      <div>
        <span>{reviews.length}</span>
        Reviews
        <ReviewList reviews={fourReviews} />
        <div>
          <button type="button">1</button>
          <button type="button">2</button>
          <button type="button">3</button>
          <button type="button">4</button>
        </div>
      </div>
    );
  }
}

App.defaultProps = {
  reviews: PropTypes.string,
};

export default App;
