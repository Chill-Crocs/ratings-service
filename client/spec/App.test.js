import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import axios from 'axios';
import React from 'react';

import App from '../components/App';

// const flushPromises = () => new Promise((resolve) => setImmediate(resolve));
const wrapper = shallow(<App />);
const wrapperInstance = wrapper.instance();
const e = { preventDefault: jest.fn() };
// jest.mock('axios');

describe('App testing...', () => {
  test('matches snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('Current page starts at 1', async () => {
    expect(wrapper.state('currentPage')).toEqual(1);
  });

  test('Increases currentPage from 1 to 2, when button is clicked', () => {
    wrapperInstance.incrementPage(e);
    expect(wrapper.state('currentPage')).toBe(2);
  });

  test('Descreases currentPage from 2 to 1, when button is clicked', () => {
    wrapperInstance.decrementPage(e);
    expect(wrapper.state('currentPage')).toBe(1);
  });

  test('Should send get request', async () => {
    const resp = {
      data: {
        _id: 1,
        shopReviews: [
          {
            _id: 1,
            avatar: 'string',
            name: 'string',
            date: 'string',
            rating: 5,
            description: 'string',
            imageUrl: 'string',
            recommended: 332,
            purchasedItem: 'string',
            shopImage: 'string',
          },
        ],
        total: 100,
      },
    };

    const getSpySuccess = jest.spyOn(axios, 'get')
      .mockResolvedValueOnce(resp)
    await wrapper.instance().componentDidMount();
    expect(getSpySuccess).toHaveBeenCalledTimes(1);
  });
});
