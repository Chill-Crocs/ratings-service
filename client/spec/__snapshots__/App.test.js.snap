// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App testing... matches snapshot 1`] = `
<div>
  <span>
    0
  </span>
  Reviews
  <ReviewList
    reviews={Array []}
  />
  <div>
    <button
      type="button"
    >
      1
    </button>
    <button
      type="button"
    >
      2
    </button>
    <button
      type="button"
    >
      3
    </button>
    <button
      type="button"
    >
      4
    </button>
  </div>
</div>
`;
